etc/syntax/reference-files/suite.rc

```diff
@@ -0,0 +1,115 @@
+# SYNTAX numbering matches the table in https://github.com/cylc/cylc/issues/2752
+
+# SYNTAX: 1.
+
+# SYNTAX: 1.1
+# keys and values should display differently
+# sections should be highlighted
+[a]
+    b = c
+    [[d]]
+        -j = k  # keys can have hyphens
+        j j = k  # and spaces
+        g = h=i  # values can have equals
+        e = "f"  # and quotes
+        e = "${HOME} `$HOME`"  # and funny characters
+    [[l, m, n]]  # sections can be contain commas
+        [[[o]]]  # sections can be deeply nested
+
+# SYNTAX: 1.2
+l=m  # no comment
+
```

Comment:
Let's also explicitly put down single-line block comments as a separate case.

Reply:
Already covered in line 19?

Reply:
All the explanatory comments are samples, such as that one, yes, but I personally would add in e.g. ``# a block comment`` to list it as a categorical case. If you think it too trifle just leave this as is, though.